_posts/en/newsletters/2022-06-15-newsletter.md

```diff
@@ -0,0 +1,281 @@
+
+    - *Using short ids:* several developers suggested using
+      [BIP152][]-style short identifiers (ids).  Zhao [explained][zhao
+      sids] that short ids make sense for block relay where nodes first
+      validate a new block's proof of work (which is expensive to
+      create), so it would be expensive for an attacker to abuse the
+      mechanism to waste node resources.  But, for relay where no
+      significant expense is involved, short ids can be abused over and
+      over again to potentially create a denial-of-service attack.
+
+    - *Non-standard parents:* Suhas Daftuar [describes][daftuar repeat]
+      a scenario where a node implementing package relay can end up
+      repeatedly requesting the same package.  This is would be
+      especially likely to happen when relay policy differs between
+      older and newer nodes, such as after a soft fork is activated.
+
+    - *Challenges of a block hash beacon:* Daftuar also notes that one
+      feature of the proposal may cause problems for other software.
+      The current draft BIP has packages include the node's current hash
+      of the latest block on the block chain.  This allows the receiving
+      peer to ignore a package if it's from an earlier block (or
+      alternative chain), in which case the package may not work with
+      the receiving peer's current mempool.  However, Daftuar notes that
+      there's probably a lot of software that sends transactions---and
+      which may eventually like to send packages---which doesn't keep
+      track of the current chain tip hash.
+
+- **Summary of LN developer meeting:** Olaoluwa Osuntokun provided a
+  [detailed summary][osuntokun summary] of the LN dev meeting in Oakland
+  last week.  Topics covered included:
+
+    - *Taproot-based LN channels:* the first step of moving LN to full
+      use of [taproot's][topic taproot] features, with later steps
+      likely to include support for [PTLCs][topic ptlc] (see also
+      [Newsletter #164][news164 taproot ln]).
+
+    - *Tapscript and MuSig2:* as part of the switch to taproot-based
+      channels, there is a need to covert existing scripts to tapscript
+      in the way that makes the most efficient use of block space.
+      There's also a desire to use [MuSig2][topic musig] for creating
+      signatures in all the places where both signers are expected to
+      act cooperatively.  Both of these need to be implemented and
+      tested to ensure they work as expected.
+
+    - *Recursive MuSig2:* a simple implementation of MuSig2 can allow
+      Alice and Bob to jointly create a single signature.  Recursive
+      MuSig2 would allow, for example, Alice to create her part of the
+      signature using both her hot wallet and a hardware signing
+      device.  It was discussed how to design LN's use of MuSig2 to
+      ensure recursive MuSig2 was available.  Also the security of
+      recursive of MuSig2 was discussed.
```

Comment:
```suggestion
      recursive MuSig2 was discussed.
```

Reply:
I haven't heard of "recursive MuSig2" before. How is that different from other ways of constructing an aggregated public key and signatures using MuSig?

Reply:
Adding details.  Thanks!